File: test_codebase/csharp/sql_injection.cs
using System.Data.SqlClient;

public class DatabaseHelper
{
    public SqlDataReader GetUser(string username)
    {
        var conn = new SqlConnection("Server=localhost;Database=test;User Id=user;Password=pass;");
        // CWE-89: SQL Injection
        string query = $"SELECT * FROM users WHERE username = '{username}'";
        var cmd = new SqlCommand(query, conn);
        return cmd.ExecuteReader();
    }
}

File: test_codebase/go/sql_injection.go
package main

import (
    "database/sql"
    "fmt"
)

func getUser(db *sql.DB, username string) (*sql.Row, error) {
    // CWE-89: SQL Injection
    query := fmt.Sprintf("SELECT * FROM users WHERE username = '%s'", username)
    return db.QueryRow(query), nil
}